docs/checkit.js

```diff
@@ -130,6 +130,128 @@ Calendar.prototype.generateMonthObjects = function (startDate, endDate) {
+};
+
+CalendarAnalyzer.prototype.getTotalCalendarDays = function () {
+    // Returns the total number of active days in the calendar
+    "use strict";
+    var endDate = moment(this.calState.endDateString, "YYYYMMDD"),
+        startDate = moment(this.calState.startDateString, "YYYYMMDD"),
+        totalDays = endDate.diff(startDate, 'days');
+    return totalDays;
+};
+
+CalendarAnalyzer.prototype.getNumberOfUnchecked = function (calObj) {
+    // Returns number of unchecked days in a calendar
+    "use strict";
+    return this.getTotalCalendarDays(calObj) - this.getNumberOfChecked(calObj);
+};
+
+CalendarAnalyzer.prototype.getCheckedDaysStreak = function () {
+    // Returns the longest streak of checked days
+    "use strict";
+    // if checkedDays is an empty object, return 0;
+    if (jQuery.isEmptyObject(this.calState.checkedDays)) {
+        return 0;
+    }
+    
+    var currentStreak = [],
+        arrayOfStreakArrays = [],
+        checkedDaysArray = [],
+        key,
+        lastDate,
+        checkedDaysArrayLength,
+        i,
+        currentDay,
+        previousDay,
+        dayDiff,
+        largestArrayIndex;
+    
+    // place checkedDays object's keys into array and sort the array.
+    for (key in this.calState.checkedDays) {
+        if (this.calState.checkedDays.hasOwnProperty(key)) {
+            checkedDaysArray.push(key);
+        }
+    }
+    checkedDaysArray.sort();
+
+    checkedDaysArrayLength = checkedDaysArray.length;
+    
+    // for each checked day
+    for (i = 0; i < checkedDaysArrayLength; i++) {
+        
+        if (i === 0) {
+            lastDate = checkedDaysArray[i];
+            currentStreak.push(lastDate);
+            i++;
+        }
+        
+        // if the next element is the day after lastDate, add it to currentStreak
+        // if not, then add currentStreak to arrayOfStreakArrays.
+        currentDay = moment(checkedDaysArray[i], "YYYYMMDD");
+        previousDay = moment(checkedDaysArray[i - 1], "YYYYMMDD");
+        
+        dayDiff = currentDay.diff(previousDay, 'days');
+
+        if (currentDay.diff(previousDay, 'days') === 1) {
+            currentStreak.push(checkedDaysArray[i]);
+        } else {
+            arrayOfStreakArrays.push(currentStreak);
+            currentStreak = [];
+            currentStreak.push(checkedDaysArray[i]);
+        }
+        
+        // if on last element, push the currentStreak into the array of streaks
+        if (i === checkedDaysArrayLength - 1) {
+            arrayOfStreakArrays.push(currentStreak);
+        }
+        
+    }
+    
+    // Find the array with the most elements in the arrayOfStreakArrays
+    largestArrayIndex = arrayOfStreakArrays.reduce(function (maxI, el, i, arr) {
+        return el.length > arr[maxI].length ? i : maxI;
+    }, 0);
+   
+    return arrayOfStreakArrays[largestArrayIndex].length;
+};
+
+CalendarAnalyzer.prototype.getUncheckedDaysStreak = function () {
```

Comment:
How is this simpler without using the page's HTML? I mean, calendarState does not have an uncheckedDays object. 
You have the startDate and the endDate and the days that are checked. So then you could maybe go through all the dates between startDate and endDate and check if they are in checkedDays and if they are not put them in an unCheckedDays array, and then look for the streak. That doesn't sound simpler lol